swap/src/protocol/alice/swap.rs

```diff
@@ -279,27 +279,68 @@ where
             ExpiredTimelocks::None { .. } => {
                 let tx_lock_status = bitcoin_wallet.subscribe_to(state3.tx_lock.clone()).await;
                 match state3.signed_redeem_transaction(*encrypted_signature) {
-                    // TODO: We should retry publishing the redeem transaction if it fails
-                    Ok(tx) => match bitcoin_wallet.broadcast(tx, "redeem").await {
-                        Ok((_, subscription)) => match subscription.wait_until_seen().await {
-                            Ok(_) => AliceState::BtcRedeemTransactionPublished { state3 },
-                            Err(e) => {
-                                bail!("Waiting for Bitcoin redeem transaction to be in mempool failed with {}! The redeem transaction was published, but it is not ensured that the transaction was included! You're screwed.", e)
+                    Ok(tx) => {
+                        // We will retry indefinitely to publish the redeem transaction, until the cancel timelock expires
+                        // We might not be able to publish the redeem transaction on the first try due to any number of reasons
+                        let backoff = backoff::ExponentialBackoffBuilder::new()
+                            .with_max_elapsed_time(None)
+                            .with_max_interval(Duration::from_secs(60))
+                            .build();
+
+                        match backoff::future::retry(backoff, || async {
+                            // If the cancel timelock is expired, we do not need to publish anymore
+                            // We cannot use a tokio::select! here because this is not cancellation safe
+                            if !matches!(
+                                state3.expired_timelocks(bitcoin_wallet).await?,
+                                ExpiredTimelocks::None { .. }
+                            ) {
```

Comment:
might be worth adding a log here

Reply:
When this happens we stop retrying (becuase we return `Ok(None)`) and enter the match branch below which logs an error:
```rust
// Cancel timelock expired before we could publish the redeem transaction
Ok(None) => {
    tracing::error!("We were unable to publish the redeem transaction before the timelock expired.");


    AliceState::CancelTimelockExpired {
        monero_wallet_restore_blockheight,
            transfer_proof,
            state3,
    }
}
```